lib/Doctrine/DBAL/Driver/Connection.php

```diff
@@ -19,16 +20,18 @@
      *
      * @param string $prepareString
      *
-     * @return \Doctrine\DBAL\Driver\Statement
+     * @return Statement
      */
     public function prepare($prepareString);
 
     /**
      * Executes an SQL statement, returning a result set as a Statement object.
      *
-     * @return \Doctrine\DBAL\Driver\Statement
+     * @return Statement
+     *
+     * @throws DBALException
      */
-    public function query();
+    public function query(string $sql);
```

Comment:
I didn't add the return type since the wrapper connection will have to declare that it returns a driver connection. This will likely change in the future.

Reply:
@morozov I didn't understand this comment: if we change this in future, wouldn't it be better to let this become a loud BC break via a change of the signature?